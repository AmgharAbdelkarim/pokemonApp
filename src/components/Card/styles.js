import styled from "styled-components";

export const CardContainer = styled.div`
  border-radius: 8px;
  background-color: #ffffff;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.25);
  > div {
    padding: 10px;
  }
  p {
    margin: 0;
  }

  .Card_img {
    text-align: center;
  }

  .Card_types {
    display: flex;
    justify-content: center;
    cursor: pointer;
  }

  .Card_type {
    padding: 5px 10px;
    margin: 10px 10px 10px 0;
    border-radius: 5px;
    background: cyan;
    color: #fff;
  }

  .Card_name {
    text-align: center;
    text-transform: capitalize;
    font-weight: 800;
    > p {
      padding: 10px;
    }
  }

  .Card_info {
    > div {
      display: flex;
      justify-content: space-between;
    }
  }

  .Card_data {
    padding-bottom: 10px;
    > p {
      padding: 10px;
    }
  }

  .Card_data .title {
    font-weight: 700;
    > p {
      padding: 10px;
    }
  }
`;
