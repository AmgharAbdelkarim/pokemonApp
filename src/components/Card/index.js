import React from 'react';
import {CardContainer} from 'components/Card/styles';
const Card = ({pokemon, clickHandler}) => {
  return (
    <CardContainer>
      <div className="Card_img">
        <img src={pokemon.sprites.front_default} alt="" />
      </div>
      <div className="Card_name">{pokemon.name}</div>
      <div className="Card_types">
        {pokemon.types.map((type) => {
          return (
            <div
              aria-hidden="true"
              className="Card_type"
              onClick={() => clickHandler(`/type/${type.type.name}`)}
            >
              {type.type.name}
            </div>
          );
        })}
      </div>
      <div className="Card_info">
        <div className="Card_data Card_data-weight">
          <p className="title">Weight</p>
          <p>{pokemon.weight}</p>
        </div>
        <div className="Card_data Card_data-weight">
          <p className="title">Height</p>
          <p>{pokemon.height}</p>
        </div>
        <div className="Card_data Card_data-ability">
          <p className="title">Ability</p>
          <p>{pokemon.abilities[0].ability.name}</p>
        </div>
      </div>
    </CardContainer>
  );
};

export default Card;
