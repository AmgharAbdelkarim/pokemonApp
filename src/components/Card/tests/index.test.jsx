import { cleanup, render } from "@testing-library/react";

import Card from "components/Card";

afterEach(cleanup);

describe("Card", () => {
  it("should Card render correctly", () => {
    const props = {
      clickHandler: jest.fn(),
      pokemon: {
        name: "fake name",
        weight: 124,
        height: 12,
        abilities: [
          {
            ability: {
              name: "water-absorb",
              url: "https://pokeapi.co/api/v2/ability/11/",
            },
            is_hidden: false,
            slot: 1,
          },
          {
            ability: {
              name: "damp",
              url: "https://pokeapi.co/api/v2/ability/6/",
            },
            is_hidden: false,
            slot: 2,
          },
          {
            ability: {
              name: "swift-swim",
              url: "https://pokeapi.co/api/v2/ability/33/",
            },
            is_hidden: true,
            slot: 3,
          },
        ],
        types: [
          {
            type: {
              name: "fakeType",
            },
          },
        ],
        sprites: {
          front_default: "fake image Url",
        },
      },
    };
    const { asFragment } = render(<Card {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
