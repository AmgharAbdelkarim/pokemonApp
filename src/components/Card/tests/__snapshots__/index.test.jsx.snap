// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should Card render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ kWMDsT"
  >
    <div
      class="Card_img"
    >
      <img
        alt=""
        src="fake image Url"
      />
    </div>
    <div
      class="Card_name"
    >
      fake name
    </div>
    <div
      class="Card_types"
    >
      <div
        aria-hidden="true"
        class="Card_type"
      >
        fakeType
      </div>
    </div>
    <div
      class="Card_info"
    >
      <div
        class="Card_data Card_data-weight"
      >
        <p
          class="title"
        >
          Weight
        </p>
        <p>
          124
        </p>
      </div>
      <div
        class="Card_data Card_data-weight"
      >
        <p
          class="title"
        >
          Height
        </p>
        <p>
          12
        </p>
      </div>
      <div
        class="Card_data Card_data-ability"
      >
        <p
          class="title"
        >
          Ability
        </p>
        <p>
          water-absorb
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
